File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/EmailConfirmCode.fs
namespace Back.Types.DataBase

open System
open System.ComponentModel.DataAnnotations

type EmailConfirmCode(toEmail: string) =
    class
        let buffer: byte array = Array.zeroCreate 15
        do Random.Shared.NextBytes buffer

        [<Key>]
        member val ConfirmCode = Convert.ToBase64String buffer with get

        member val ToEmail = toEmail with get
        member val DeleteAt = DateTime.Now.AddDays 1
    end

File: AspNetCore_gdr2333_Front+Back/Back/Types/DataBase/EmailConfirmCode.cs
using System.ComponentModel.DataAnnotations;
using System.Security.Cryptography;

namespace Back.Types.DataBase;

public class EmailConfirmCode
{
    [Key]
    public string ConfirmCode { get; set; }

    public string ToEmail { get; set; }

    public long AccountId { get; set; }

    public DateTime DeleteAt { get; set; }

    public EmailConfirmCode(string toEmail, long accountId)
    {
        ToEmail = toEmail;
        ConfirmCode = Guid.NewGuid().ToString();
        DeleteAt = DateTime.Now.AddDays(1);
        AccountId = accountId;
    }

    public EmailConfirmCode()
    {
    }
}
